### ISSUE TITLE
Change hook used for maybe_insert_user

### ISSUE THREAD

**Issue by mintplugins** [easydigitaldownloads/edd-auto-register#50]
Auto register no longer registers users through Free Downloads. This is because Auto Register runs the maybe_insert_user function by hooking to the edd_insert_payment hook, which only runs during the edd_insert_payment function, which is a legacy function that Free Downloads no longer uses.

We should be able to fix this here in Auto Register by hooking to an existing payment in the `EDD_Payment::insert_payment()` method or `EDD_Payment::save()` method instead of edd_insert_payment. 

Related:
https://github.com/easydigitaldownloads/edd-free-downloads/issues/238

**Comment by SDavisMedia** [comment]
Definitely replicated on `master`. Working as expected for me on `issue/50`. :+1:

How are we using `edd_debug_log()` moving forward? I actually didn't even know about the new enhancement until a week or so ago. I see this PR uses it quite a bit in a way that would imply we'd be logging tons of stuff all over the ecosystem. I'm curious if there's a standard for its use or what.

**Comment by mintplugins** [comment]
@SDavisMedia I just added a fix for non Free Downloads payments. Both are working for me here now. Can you test with both a Free Download and a normal purchase to make sure everything works as expected?

**Comment by mintplugins** [comment]
P.S. I don't know if there is a standard for how to use edd_debug_log yet, but it's probably a good idea.

**Comment by SDavisMedia** [comment]
@mintplugins yup, it's working for me now. Saw a debug log error earlier that I was gonna mention but it looks like you caught it. All is well on my end. 👍

**Comment by mintplugins** [comment]
I changed the `isset` check to an `empty` check as per @pippinsplugins reivew on the PR and can confirm that it still works correctly for both FD and normal purchases. Thanks @SDavisMedia

**Comment by pippinsplugins** [comment]
Changes here caused https://github.com/easydigitaldownloads/edd-auto-register/issues/52.

We have to reassess this.

**Comment by mintplugins** [comment]
Since #52 is now merged, I believe this can be closed. New users are created through Free Downloads in the latest `master`.

**Comment by mintplugins** [comment]
I discussed this with @cklosowski and we decided this issue can be closed now that #52 is tested, working, and merged.

To re-cap, here is a timeline of events:
- This issue was created and the PR was submitted, merged, and released in 1.3.10
- 1.3.10 caused problems with recurring
- The WP Plugin repo for Auto Register was rolled back from 1.3.10 to 1.3.9, a couple of users were affected
- Issue #52 was opened, using `master` as the source of the branch, which included this issue's PR code
- Issue #52 was completed and merged. 
- I have tested to make sure Auto Register still works with Free Downloads, which is what this issue is really for. 
- Therefore, this issue is now being closed.
